Added checkbox support

vonNiklasson opened the pull request:
Noticed that checkboxes didn't initialize automatically. Added a few lines in the files in src. Have tested it locally but doesn't know how to add it to the test environment on the git project.

bunzli commented:
Hi @vonNiklasson 

It's a nice idea to support checkbox inputs. It would be great if you add some tests, something like this:

```javascript
describe('keepCurrentValue directive on checkbox (single)', function() {
    var html, element, scope;

    beforeEach(inject(function($rootScope, $compile) {
      html = '<input type="checkbox" keep-current-value name="bike" ng-model="bike" checked value="bike" />';

      element = angular.element(html);
      scope = $rootScope.$new();
      $compile(element)(scope);
      scope.$digest();
    }));

    describe('with tags value and not checked', function() {
      it('should set the ngModel\'s value', function(){
        expect(scope.bike).toEqual('bike');
      });
    });
  });
```

All combinations must be covered (with and without checked tag, and with and without value tag).

Also, it's not necessary to edit `dist/angular-keep-values.js`.

vonNiklasson commented:
Hello again! A while later but I've implemented a broader lookup and added tests for it. Also removed what I assumed was a leftover `console.log` in the code.

Didn't know how wide I should make the test so I implemented the most important parts of it.

Cheers,
Johan

blackjid commented:
LGTM!

blackjid commented:
@vonNiklasson could you please add an example to the example section in the Readme??

vonNiklasson commented:
Sure, I formatted it so the JS-result would appear in the same order as the HTML.

vonNiklasson commented on src/keep-current-value.js:
```diff
@@ -48,7 +60,7 @@ function keepCurrentValue() {
   function link(scope, element, attrs, controller) {
     var viewValue = getViewValueFromElement(element);
-    if ( viewValue ) {
+    if ( viewValue !== undefined ) {
```
Since the value `false` can be returned if the checkbox isn't checked and no ng-false-value is set, `viewValue` needs a more precise comparsion.
